### PR TITLE
Add schema classes for Draft4, Draft6 and Draft7

### PR DESCRIPTION
**Pull request by jhthorsen**
### Summary
This PR has sub classes for JSON-Schema draft-4, 6 and 7. I think there are more things that could be factored out from JSON::Validator, but maybe this is a good start?

### Motivation
See #209 for motivation.

### References
#209, #189.

### CODE REVIEW

File: lib/JSON/Validator/Schema/Draft7.pm [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    'iri'           => JSON::Validator::Formats->can('check_iri'),
+    'iri-reference' => JSON::Validator::Formats->can('check_iri_reference'),
+    'json-pointer'  => JSON::Validator::Formats->can('check_json_pointer'),
+    'regex'         => JSON::Validator::Formats->can('check_regex'),
+    'relative-json-pointer' =>
+      JSON::Validator::Formats->can('check_relative_json_pointer'),
+    'time'          => JSON::Validator::Formats->can('check_time'),
+    'uri'           => JSON::Validator::Formats->can('check_uri'),
+    'uri-reference' => JSON::Validator::Formats->can('check_uri_reference'),
+    'uri-reference' => JSON::Validator::Formats->can('check_uri_reference'),
```
Comment (karenetheridge) [inline lib/JSON/Validator/Schema/Draft7.pm:28]: duplicate line

Comment (jhthorsen) [reply lib/JSON/Validator/Schema/Draft7.pm]: Fixed

**Comment by karenetheridge** [comment]
Some of these changes are not correct. maxContains and minContains are not part of the draft-07 spec -- they were added in draft 2019-09. duration and uuid formats weren't added until 2019-09 either.

**Comment by jhthorsen** [comment]
Thanks!

**Pull request merged**